### PR TITLE
feat(ui): check admin status via service instead of JWT

### PR DESCRIPTION
**Pull request by liangfung**
Relates #1675 
Use 'queryMe'

### CODE REVIEW

File: ee/tabby-ui/lib/tabby/auth.tsx [hunk 1]
```diff
@@ -271,7 +254,6 @@ function useSignOut(): () => Promise<void> {
 }
 
 interface User {
```
Comment (wsxiaoys) [inline ee/tabby-ui/lib/tabby/auth.tsx:256]: nit: rename this to something like JWTPayload

File: ee/tabby-ui/app/(dashboard)/components/sidebar.tsx [hunk 2]
```diff
@@ -31,8 +31,8 @@ export interface SidebarProps {
 }
 
 export default function Sidebar({ children, className }: SidebarProps) {
-  const { data: session } = useSession()
-  const isAdmin = session?.isAdmin || false
+  const [{ data }] = useMe()
+  const isAdmin = data?.me?.isAdmin
```
Comment (wsxiaoys) [inline ee/tabby-ui/app/(dashboard)/components/sidebar.tsx:35]:
```suggestion
  const isAdmin = data?.me.isAdmin
```

`me` is always not null if data is not null?

File: ee/tabby-ui/lib/tabby/auth.tsx (hunk 1 above)
Comment (liangfung) [reply ee/tabby-ui/lib/tabby/auth.tsx]: Used `IJwtPayload` to differentiate from the `jwtPayload` exported by `jwt-decode`.

File: ee/tabby-ui/lib/tabby/auth.tsx [hunk 3]
```diff
@@ -270,8 +253,7 @@ function useSignOut(): () => Promise<void> {
   }
 }
 
-interface User {
-  isAdmin: boolean
+interface IJwtPayload {
```
Comment (wsxiaoys) [inline ee/tabby-ui/lib/tabby/auth.tsx:256]:
```suggestion
interface JwtPayload {
```

Prefer stating type directly, without `I` prefix

**Pull request merged**